src/components/graph/network-modification-tree-model.js

```diff
@@ -148,4 +151,15 @@ export default class NetworkModificationTreeModel {
         );
         return newTreeModel;
     }
+
+    setBuildingStatus() {
+        for (let i = 0; i < this.treeElements.length; i++) {
```

Comment:
why not using
isOneNodeBuilding = treeElements.first(el => el?.data?.buildStatus === 'BUILDING') !== undefined
?

Reply:
Done. Yes that's classier